File: sdk/agent-sdk-typescript/src/exceptions.ts
export class AgentIDError extends Error {
  constructor(message: string) {
    super(message);
    this.name = 'AgentIDError';
  }
}

export class ConnectError extends AgentIDError {}
export class AuthFailedError extends ConnectError {}
export class DiscoverError extends AgentIDError {}
export class InvokeError extends AgentIDError {}
export class NotAuthorizedError extends InvokeError {}
export class HitlRequiredError extends InvokeError {}
export class DelegateError extends AgentIDError {}
export class MaxDepthError extends DelegateError {}
export class PtvError extends AgentIDError {}
export class HitlError extends AgentIDError {}
export class McpError extends AgentIDError {}
export class VerifyError extends AgentIDError {}
export class SkillError extends AgentIDError {}
export class TxError extends AgentIDError {}

File: sdk/agent-sdk-python/src/agentid_sdk/exceptions.py
from __future__ import annotations


class AgentIDError(Exception):
    pass


class ConnectError(AgentIDError):
    pass


class AuthFailedError(ConnectError):
    pass


class DiscoverError(AgentIDError):
    pass


class InvokeError(AgentIDError):
    pass


class NotAuthorizedError(InvokeError):
    pass


class HitlRequiredError(InvokeError):
    pass


class DelegateError(AgentIDError):
    pass


class MaxDepthError(DelegateError):
    pass


class PtvError(AgentIDError):
    pass


class HitlError(AgentIDError):
    pass


class McpError(AgentIDError):
    pass


class VerifyError(AgentIDError):
    pass


class SkillError(AgentIDError):
    pass


class TxError(AgentIDError):
    pass